server/redeemer.go

```diff
@@ -0,0 +1,383 @@
+	return err
+}
+
+// MaxFloat returns the max float for 'sender'
+// If no local cache is available this method will remotely request
+// max float from the Redeemer server and start watching for subsequent updates from the Redeemer server
+func (r *RedeemerClient) MaxFloat(sender ethcommon.Address) (*big.Int, error) {
+	// Ensure thread-safe access to 'RedeemerClient.senders'
+	r.mu.Lock()
+
+	// Check whether local cache exists for 'sender'
+	if mf, ok := r.senders[sender]; ok && mf.maxFloat != nil {
+		r.senders[sender].lastAccess = time.Now()
+		return mf.maxFloat, nil
+	}
+
+	// Retrieve max float from Redeemer and cache it if no local cache exists
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCTimeout)
+	defer cancel()
+	mfu, err := r.rpc.MaxFloat(ctx, &net.MaxFloatReq{Sender: sender.Bytes()})
+	if err != nil {
+		return nil, err
+	}
+	mf := new(big.Int).SetBytes(mfu.MaxFloat)
+
+	// Request updates for sender from Redeemer
+	stream, err := r.rpc.MonitorMaxFloat(ctx, &net.MaxFloatReq{Sender: sender.Bytes()})
+	// By sending a message over the existing stream to the Redeemer
+	if err != nil {
+		// An error means we won't be receiving updates from the Redeemer for 'sender'
+		// So don't update the local cache and retry subscribing to updates from 'sender' on the next MaxFloat() call
+		// Return the current retrieved max float
+		glog.Error(err)
+		return mf, nil
+	}
+
+	r.senders[sender] = &remoteSender{
+		maxFloat:   mf,
+		stream:     stream,
+		done:       make(chan struct{}),
+		lastAccess: time.Now(),
+	}
+
+	// release the lock before starting a watchdog on the stream for the 'sender'
+	r.mu.Unlock()
+
+	go r.monitorMaxFloat(sender)
+
+	return mf, nil
+}
+
+// ValidateSender checks whether a sender has not recently unlocked its deposit and reserve
+func (r *RedeemerClient) ValidateSender(sender ethcommon.Address) error {
+	info, err := r.sm.GetSenderInfo(sender)
+	if err != nil {
+		return fmt.Errorf("could not get sender info for %v: %v", sender.Hex(), err)
+	}
+	maxWithdrawRound := new(big.Int).Add(r.tm.LastInitializedRound(), big.NewInt(1))
+	if info.WithdrawRound.Int64() != 0 && info.WithdrawRound.Cmp(maxWithdrawRound) != 1 {
+		return fmt.Errorf("deposit and reserve for sender %v is set to unlock soon", sender.Hex())
+	}
+	return nil
+}
+
+func (r *RedeemerClient) monitorMaxFloat(sender ethcommon.Address) {
+	r.mu.Lock()
+	s, ok := r.senders[sender]
+	r.mu.Unlock()
+	// sanity check
+	if !ok {
+		return
+	}
+
+	updateC := make(chan *net.MaxFloatUpdate)
+	errC := make(chan error)
+	go func() {
+		for {
+			update, err := s.stream.Recv()
+			if err != nil {
+				glog.Error(err)
+				if err == io.EOF {
+					errC <- err
+					return
+				}
+			} else {
+				updateC <- update
+			}
+		}
+	}()
+
+	for {
+		select {
+		case <-r.quit:
+			if err := s.stream.CloseSend(); err != nil {
+				glog.Error(err)
+			}
+			return
+		case <-s.done:
+			glog.Infof("closing stream for sender=%v", sender.Hex())
```

Comment:
Let's capitalize "Closing" for consistency since a lot of our info logs capitalize the first word in the message.

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/1513/commits/6b7ac892cd933ab613b66a1cd5ae30cd990c128e